dlib/travis/build-and-test.sh

```diff
@@ -7,11 +7,16 @@ if [ -n "${MATRIX_EVAL+set}" ]; then
     eval "${MATRIX_EVAL}"
 fi
 
+CXX_FLAGS="-std=c++11"
+if [ ! -z ${CXXFLAGS+x} ]; then
```

Comment:
Why is there that +x there?  
```suggestion
if [ ! -z ${CXXFLAGS} ]; then
```
Why not do this?

Reply:
Heh, I don't recall why I did that odd `+set` thing on the one above either :|

Reply:
Yes, that was my first try...